src/commands/init.js

```diff
@@ -83,84 +222,323 @@ Examples:
-      delete pkgJson['license']
-      delete pkgJson['repository']
+      return await initSubgraphFromContract(
+        toolbox,
+        { abi, allowSimpleName, directory, address: fromContract, network, subgraphName },
+        { commands }
+      )
+    }
 
-      // Write package.json
-      await filesystem.write(pkgJsonFilename, pkgJson, { jsonIndent: 2 })
-    } catch (e) {
-      print.error(`Failed to preconfigure the subgraph: ${e}`)
-      process.exitCode = 1
-      filesystem.remove(directory)
+    // Otherwise, take the user through the interactive form
+    let inputs = await processInitForm(toolbox, {
+      abi,
+      allowSimpleName,
+      directory,
+      address: fromContract,
+      fromExample,
+      network,
+      subgraphName,
+    })
+
+    if (inputs === undefined) {
+      process.exit(1)
       return
     }
 
-    // Reset the git repository
-    try {
-      spinner = print.spin('Creating initial commit')
-      await filesystem.remove(filesystem.path(directory, '.git'))
-      await system.run('git init', { cwd: directory })
-      await system.run('git add --all', { cwd: directory })
-      await system.run('git commit -m "Initial commit"', {
-        cwd: directory,
+    print.info('———')
+
+    if (fromExample) {
+      await initSubgraphFromExample(toolbox, {
+        subgraphName: inputs.subgraphName,
+        directory: inputs.directory,
       })
-      spinner.stop()
-    } catch (e) {
-      spinner.fail(
-        `Failed to initialize the subgraph directory ${relativeDirectory}: ${e}`
+    } else {
+      await initSubgraphFromContract(
+        toolbox,
+        {
+          allowSimpleName,
+          subgraphName: inputs.subgraphName,
+          directory: inputs.directory,
+          abi: inputs.abi,
+          network: inputs.network,
+          address: inputs.address,
+        },
+        { commands }
       )
-      process.exitCode = 1
-      filesystem.remove(directory)
-      return
     }
+  },
+}
+
+const revalidateSubgraphName = async (toolbox, subgraphName, { allowSimpleName }) => {
+  // Fail if the subgraph name is invalid
+  try {
+    validateSubgraphName(subgraphName, { allowSimpleName })
+    return true
+  } catch (e) {
+    toolbox.print.error(`${e.message}
+
+  Examples:
 
-    print.success(
-      `Subgraph "${subgraphName}" created in ${relativeDirectory}${filesystem.separator}`
-    )
-    print.info(`
-Next steps:
-    
-  1. Run \`graph auth https://api.thegraph.com/deploy/ <your access token>\`
+    $ graph init ${os.userInfo().username}/${subgraphName}
+    $ graph init ${subgraphName} --allow-simple-name`)
+    return false
+  }
+}
+
+const initRepository = async (toolbox, directory) =>
+  await withSpinner(
+    `Initialize subgraph repository`,
+    `Failed to initialize subgraph repository`,
+    async spinner => {
+      let gitDir = path.join(directory, '.git')
+      if (toolbox.filesystem.exists(gitDir)) {
+        await toolbox.filesystem.remove(gitDir)
```

Comment:
How could the git directory already exist, doesn't graph <strike>autodave</strike> init always start from an empty dir?

Reply:
With `--from-contract` yes, but not with `--from-example` (where it clones the example subgraph). This code is used in both cases.

Reply:
Ah ok, so maybe comment that this is removing the .git in the `--from-example` case.

Reply:
Done